File: Python/1TwoSum.py
# Just use a dict to track each number and index pair and use it for fast lookup.

class Solution(object):
    def twoSum(self, nums, target):
        d = {}
        for index, val in enumerate(nums):
            if (target - val) in d:
                return [index, d[target - val]]
            else:
                d[val] = index
        return

File: Java/1TwoSum.java
/*
 * We create a HashMap containing each number and its index for fast lookup.
 */

public class 1TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> h = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (h.containsKey(target - nums[i])) {
                int[] ans = {i, h.get(target - nums[i])};
                return ans;
            } else {
                h.put(nums[i], i);
            }
        }
        return null;
    }
}
